website/docs-scraper/extract-content-from-markdown.mjs

```diff
@@ -0,0 +1,217 @@
+        }
+      });
+    });
+  };
+
+  const wcagListMapper = () => (tree) => {
+    // the `<Doc::WcagList @criteriaList={{array "1.1.1" "2.2.2" />`
+    // becomes a `text` node within a `paragraph` node
+    // so we have to do some magic here... ¯\_(ツ)_/¯
+    visit(tree, 'text', (node) => {
+      const match = node.value.match(
+        /<doc-wcag-list @criteriaList={{array .*}} \/>/i
+      );
+      if (match) {
+        const hbsAST = handlebars.parse(node.value);
+        const criteriaList = hbsAST.body.filter(
+          (node) =>
+            node?.type === 'MustacheStatement' &&
+            node?.path?.original === 'array'
+        );
+        let criteriaIdentifiers;
+        if (
+          criteriaList &&
+          criteriaList.length > 0 &&
+          criteriaList[0].params &&
+          criteriaList[0].params.length > 0
+        ) {
+          criteriaIdentifiers = criteriaList[0].params.reduce((acc, param) => {
+            if (
+              param.type === 'StringLiteral' &&
+              Object.keys(WCAG_CRITERIA).includes(param.value)
+            ) {
+              acc.push(param.value);
+            }
+            return acc;
+          }, []);
+        }
+        if (criteriaIdentifiers.length > 0) {
+          wcagLists.push({
+            criteria: criteriaIdentifiers.map((criterion) => {
+              return _.pick(WCAG_CRITERIA[criterion], [
+                'number',
+                'title',
+                'description',
+              ]);
+            }),
+            hierarchy: node.hierarchy,
+          });
+        }
+      }
+    });
+  };
+
+  // we need to convert the `<Doc::***>` components to web-components-like code (HTML compatible)
+  const standardizedContent = replaceDocTags(markdownContent);
+
+  // processing pipeline
+  await remark()
+    // process custom (showdown.js) format
+    .use(remarkProcessCustomImageFormat)
+    // pre-emptively remove any comment, just in case
+    .use(remarkRemoveComments)
+    // pre-emptively remove any comment, just in case
+    .use(remarkRemoveCodeBlocks)
+    // remove/strip all the custom entities (custom markdown, useless ember components, etc)
+    .use(remarkStripHeliosContentBlocksDelimiters)
+    // .use(remarkStripHeliosReleaseNotesMetadata)
+    // convert markdown to HTML (TODO do we need it? what happens if we remove this?)
+    .use(remarkHtml, { sanitize: remarkHtmlSanitise })
+    // interpret special GFM markdown format
+    .use(remarkGfm)
+    // associate to each node the hierarchy in terms of headings level
+    .use(setNodesHierarchy)
+    // parse and index "doc" (ember) nodes for special components
+    .use(wcagListMapper)
+    .use(componentApiMapper)
+    // remove handlebars expressions (`{{...}}`) not that are not useful anymore, so they don't pollute the paragraphs
+    .use(remarkStripHeliosHandlebarsExpressions)
+    // remove empty paragraphs
+    // notice: done later in the process, more efficient than doing it here
```

Comment:
should this get removed then?

Reply:
I want to leave it here for the time being, will be cleaned up in the second part of the project when we try to clean up even more the indexed content.